src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateMethodSymbol.cs

```diff
@@ -301,15 +301,27 @@ internal override LexicalSortKey GetLexicalSortKey()
 
             internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, DiagnosticBag diagnostics)
             {
+                Location getReturnTypeLocation()
+                {
+                    var syntax = (DelegateDeclarationSyntax)SyntaxRef.GetSyntax();
```

Comment:
>(DelegateDeclarationSyntax)SyntaxRef.GetSyntax(); [](start = 33, length = 49)

Consider asserting that this cast is always safe before this local function.

Reply:
Added assert.